CMakeLists.txt

```diff
@@ -99,6 +104,10 @@ zephyr_compile_options(
   ${NOSTDINC_F}
 )
 
+if(${ZEPHYR_TOOLCHAIN_VARIANT} STREQUAL "xcc")
+  zephyr_compile_options(-fms-extensions)
```

Comment:
should i employ a method similar to the one adopted for OPTIMIZE_FOR_SIZE_FLAG flag? i.e. define a new flag, something like FMS_EXTENSIONS in toolchain build script and then include it under zephyr_compile_options in the toplevel CMakeLists.txt.
Another option is to use zephyr_cc_option directly. It will fail for gcc but work for xcc. Let me know your thoughts.

Reply:
Hi, I don't think -fms-extensions will fail for gcc actually.

I believe the cleanest solution will be to do

  list(APPEND TOOLCHAIN_C_FLAGS -fms-extensions)

from the toolchain build script.

Reply:
Well.. that was my preferred way as well in the beginning when i started providing this support. It just doesn't work. It just doesn't seem to take effect at all. That's the reason why i modified the toplevel CMakeLists.txt.

Reply:
Right, so the reason TOOLCHAIN_C_FLAGS is taking effect on 'arm', but not on 'xtensa' is that arm is doing

zephyr_compile_options(
  ${TOOLCHAIN_C_FLAGS}
  )

I suspect that this is undesirable, the toplevel CMakeLists.txt should be doing

zephyr_compile_options(
  ${TOOLCHAIN_C_FLAGS}
  )

so that it is not architecture-dependent whether toolchain flags are used or not.

Anyway, for this particular PR we need to get rid of

+if(${ZEPHYR_TOOLCHAIN_VARIANT} STREQUAL "xcc")

in the toplevel CMakeLists.txt file.

This can be done by copying what 'arm' is doing, and executing

 zephyr_compile_options(${TOOLCHAIN_C_FLAGS})

in arch/xtensa/CMakeLists.txt

or by adding TOOLCHAIN_C_FLAGS in the toplevel CMakeLists.txt and removing it from arch/{arm,posix}.

Reply:
Thanks for the info. I have modified the toplevel CMakeLists.txt and it is working fine.